lighthouse-core/scripts/generate-bundlephobia-database.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+/* eslint-disable no-console */
+
+/** @typedef {{name: string, version: string, gzip: number, description: string, repository: string}} BundlePhobiaLibrary */
+
+const fs = require('fs');
+const exec = require('child_process').exec;
+
+/** @type string[] */
+const libraries = require('../audits/byte-efficiency/library-suggestions.js').suggestions.flat();
+const databasePath = '../audits/byte-efficiency/bundlephobia-database.json';
+
+/** @type {Record<string, Record<'lastScraped', number|string> | Record<string, BundlePhobiaLibrary>>} */
+let database = {};
+if (fs.existsSync(databasePath)) {
+  database = require(databasePath);
+}
+
+/**
+ * Returns true if this library has been scraped from BundlePhobia in the past week.
+ * @param {string} library
+ * @return {boolean}
+ */
+function hasBeenRecentlyScraped(library) {
+  if (!database[library] || database[library].lastScraped === 'Error') return false;
+  return (Date.now() - database[library].lastScraped) / (1000 * 60 * 60 * 24.0) < 7;
+}
+
+/**
+ * Returns true if the object represents valid BundlePhobia JSON.
+ * The version string must not match this false-positive expression: '{number} packages'.
+ * @param {any} library
+ * @return {boolean}
+ */
+function validateLibraryObject(library) {
+  return library.hasOwnProperty('name') &&
+    library.hasOwnProperty('size') &&
+    library.hasOwnProperty('gzip') &&
+    library.hasOwnProperty('description') &&
+    library.hasOwnProperty('repository') &&
+    library.hasOwnProperty('version') &&
+    !library.version.match(/^([0-9]+) packages$/);
+}
+
+/**
+ * Save BundlePhobia stats for a given npm library to the database.
+ * @param {string} library
+ * @param {number} index
+ */
+async function collectLibraryStats(library, index) {
+  return new Promise((resolve, reject) => {
+    console.log(`\n◉ (${index}/${libraries.length}) ${library} `);
+
+    if (hasBeenRecentlyScraped(library)) {
+      console.log(`   ❕ Skipping`);
+      resolve();
+      return;
+    }
+
+    exec(`bundle-phobia ${library} -j -r`, (error, stdout) => {
```

Comment:
bundle phobia CLI exposes a node API to get this information directly as an object without going through this child process business

https://github.com/AdrieanKhisbe/bundle-phobia-cli/blob/f2449b89ea03e99f94662b697e2b8f3ba0fb2024/src/index.js#L1-L4

will that work for the data you need?

Reply:
or good go straight to the API it's not that bad to use :)

```js
const fetch = require('isomorphic-fetch')
const packageData = await fetch('https://bundlephobia.com/api/size?package=moment').then(r => r.json())

console.log(packageData)

{ assets: [ { gzip: 72054, name: 'main', size: 287596, type: 'js' } ],
  dependencyCount: 0,
  dependencySizes:
   [ { approximateSize: 362248, name: 'moment' },
     { approximateSize: 352, name: 'webpack' } ],
  description: 'Parse, validate, manipulate, and display dates',
  gzip: 72054,
  hasJSModule: false,
  hasJSNext: './dist/moment.js',
  hasSideEffects: true,
  name: 'moment',
  repository: 'https://github.com/moment/moment.git',
  scoped: false,
  size: 287596,
  version: '2.27.0' }
```

Reply:
Thanks :) moved to the API